app/main.py

```diff
@@ -1,13 +1,24 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+    return {
+        ("name" if key == "code" else "source" if key == "filename" else
+         "message" if key == "text" else tuple(key.split("_"))[0]):
+        ("flake8" if key == "filename" else value)
+        for key, value in error.items() if key != "physical_line"
```

Comment:
not readable code. please, use static key names and choose the correct fields from the dictionary for those keys. for example:
```python
"name": error["code"]
```